### PR TITLE
Overall optimization (it should work without testing but testing would be better)

### PR DESCRIPTION
**Pull request by EdoTM**
Overall optimization:
- Replaced some deprecated Gmod functions with new ones (e.g. `SetNetworked...` with `SetNW...`, `surface.ScreenWidth()` with `ScrW()`,`table.getn(tbl)` with `#tbl`, etc.)
- Deleted empty if and for statements
- Replaced user messages (and server messages) with Net functions
- Removed unused local variables
- Replaced old dependency `SetNetworkedBeamString` with `SetNWString`
- Tidy code

I didn't actually edit anything else other than these things.

### CODE REVIEW

**Comment by TomyLobo** [comment]
Well anyone can replace one function call with another, so just to be sure you understand what you're doing, describe what `SetNetworkedBeamString` is.

**Comment by thegrb93** [comment]
SetNetworkedBeamString are just wire's implementation of SetNWString that was removed a long time ago.

**Comment by EdoTM** [comment]
Networked Beams were an attempt to optimize the data flow between server and client when treating large data (there's also a description [**here**](https://github.com/wiremod/wire/blob/e0224feb491621efe7c5d41849328df601a1a6cc/lua/wire/beam_netvars.lua), a more-than-10-years-old file) but they eventually got removed (along with the NW Beam library) by AbigailBuccaneer in 2015.
Some wire extras items (such as doors and door controllers) still use these, and they do generate errors when you try to use them (actually, in some cases you **can't** use them due to these errors). 

A fix would be changing `SetNetworkedBeamString` to `SetNWString`, as it has already been done in #74. I personally tried doors and door controllers, and with this fix they work smoothly without a single error.

**Comment by thegrb93** [comment]
I'll have time to review tomorrow.

**Comment by TomyLobo** [comment]
@thegrb93 You do realize that you giving any kind of hint on what beams are kinda defeats the point of that test :)
@EdoTM passed with flying colors, though :)

**Comment by thegrb93** [comment]
Oh, thought you were asking rather than testing

**Comment by EdoTM** [comment]
I'm now removing unused `net.Write*`/`Read*`, and I noticed that in `wire_modular_panel.lua` there's a strange behavior in usermessages and servermessages. On [line 325](https://github.com/wiremod/wire-extras/blob/653a1f5830071ce1f88cae57a211aa8948c50db9/lua/weapons/gmod_tool/stools/wire_modular_panel.lua#L325) of the file that is currently on the repository, a ***user***message named `smsgModularPanel` is initialized (the name is the same of the servermessage defined on [line 204](https://github.com/wiremod/wire-extras/blob/653a1f5830071ce1f88cae57a211aa8948c50db9/lua/weapons/gmod_tool/stools/wire_modular_panel.lua#L204). They are correlated, but I don't know if the fact of them having an identical name is actually intentional), but there's no `usermessage.Hook("smsgModularPanel", ...)` or `umsgModularPanel` in any file. 
Am I missing something?

**Review by thegrb93 — CHANGES_REQUESTED** [review]

**Comment by thegrb93** [comment]
Looks like a 1 to 1 conversion and looks fine. If you wouldn't mind just cleaning up the unneeded comments and revert the GetConVar change, it will be perfect.

File: lua/entities/base_wdr_entity/init.lua [hunk 1]
```diff
@@ -187,7 +187,7 @@ function ENT:Think()
 		end
 
 		for k, v in pairs(txs) do
-			local dist = self:GetPos():Distance(v:GetPos()) / GetConVarNumber("sv_wdrk_scale")
+			local dist = self:GetPos():Distance(v:GetPos()) / GetConVar("sv_wdrk_scale"):GetInt()
```
[the remark this answers is not in the document]
Comment (EdoTM) [reply lua/entities/base_wdr_entity/init.lua]: Done!

File: lua/entities/gmod_wire_adv_hudindicator/cl_init.lua [hunk 2]
```diff
@@ -5,8 +5,8 @@ ENT.RenderGroup 		= RENDERGROUP_BOTH
 
 local advhudindicators = {}
 // Default HUD x/y
-local hudx = 0
-local hudy = 0
+//local hudx = 0
+//local hudy = 0
```
[the remark this answers is not in the document]
Comment (EdoTM) [reply lua/entities/gmod_wire_adv_hudindicator/cl_init.lua]: I was leaving them to highlight what I was changing, I now removed them

**Comment by EdoTM** [comment]
Removed unnecessary comments and reverted `GetConVar(...):GetInt()` with `GetConVarNumber(...)`

**Comment by thegrb93** [comment]
Thanks. Looks good

**Pull request merged**